core/domain/question_jobs_one_off.py

```diff
@@ -140,3 +142,105 @@ def reduce(key, values):
                 sum(ast.literal_eval(v) for v in values))])
         else:
             yield (key, values)
+
+
+class MissingQuestionMigrationOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """This job is used to delete question commit log models for which question
+    models are missing.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [question_models.QuestionCommitLogEntryModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        question = question_services.get_question_by_id(
+            item.question_id, strict=False)
+        if question is None:
+            item.delete()
+            yield ('Question Commit Model deleted', 1)
```

Comment:
This is actually kind of a "failure" case so let's output the deleted IDs.

Reply:
Done.